libcodechecker/output_formatters.py

```diff
@@ -0,0 +1,186 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+"""
+Contains functions to format and pretty-print data from two-dimensional arrays.
+"""
+
+import json
+
+# The supported formats the users should specify. (This is not an exhaustive
+# list of ALL formats available.)
+USER_FORMATS = ['rows', 'table', 'csv', 'json']
+
+
+def twodim_to_str(format, keys, rows):
+    """
+    Converts the given two-dimensional array (with the specified keys)
+    to the given format.
+    """
+
+    all_rows = rows
+    if keys is not None and len(keys) > 0:
+        all_rows = [keys] + rows
+
+    if format == 'rows':
+        return twodim_to_rows(rows)
```

Comment:
Expected to throw. It is the client code's responsibility to prepare a well-formatted twodim.